test/OpenTelemetry.Tests/Metrics/MetricExporterTests.cs

```diff
@@ -44,22 +44,25 @@ public void FlushMetricExporterTest(ExportModes mode)
                     break;
             }
 
+            var reader = new BaseExportingMetricReader(exporter);
             using var meterProvider = Sdk.CreateMeterProviderBuilder()
-                .AddMetricReader(new BaseExportingMetricReader(exporter))
+                .AddMetricReader(reader)
                 .Build();
 
-            var result = meterProvider.ForceFlush();
-
             switch (mode)
             {
                 case ExportModes.Push:
-                    Assert.True(result);
+                    Assert.True(reader.Collect());
+                    Assert.True(meterProvider.ForceFlush());
                     break;
                 case ExportModes.Pull:
-                    Assert.False(result);
+                    Assert.False(reader.Collect());
+                    Assert.False(meterProvider.ForceFlush());
+                    Assert.True(exporter.Collect(reader));
```

Comment:
This extension method on `BaseExporter<Metrics>` allows the pull exporter to collect metrics.

Reply:
Need more design discussion on this, please refer to https://github.com/open-telemetry/opentelemetry-dotnet/pull/2389#discussion_r712708670.